TFEngine.py

```diff
@@ -1752,22 +1752,25 @@ def search_single_seq(self, source, output_layer_name=None):
     dataset.init_seq_order(epoch=1)
     return self.search_single(dataset=dataset, seq_idx=0, output_layer_name=output_layer_name)
 
-  def search_single_bpe_to_bpe_seq(self, source, source_bpe, target_bpe, output_layer_name=None):
+  def search_single_string_to_string_seq(self, source, output_layer_name=None):
     """
     :param str source: source as a string
-    :param GeneratingDataset.BytePairEncoding source_bpe:
-    :param GeneratingDataset.BytePairEncoding target_bpe:
     :param str|None output_layer_name: e.g. "output". if not set, will read from config "search_output_layer"
     :return: list of all hyps, which is a tuple of score and string
     :rtype: list[(float,str)]
     """
-    assert source_bpe.num_labels == self.network.extern_data.data["data"].dim
-    assert target_bpe.num_labels == self.network.extern_data.data["classes"].dim
-    source_seq_list = source_bpe.get_seq(source)
+    self._checked_uninitialized_vars = True
```

Comment:
A final question: Why was this needed here? I could imagine that this could cause troubles in some cases.